commonItems/Color.cs

```diff
@@ -0,0 +1,296 @@
+                    throw new Exception("Color has wrong number of digits");
+                }
+                var r = int.Parse(hex.Substring(0, 2), System.Globalization.NumberStyles.HexNumber);
+                var g = int.Parse(hex.Substring(2, 2), System.Globalization.NumberStyles.HexNumber);
+                var b = int.Parse(hex.Substring(4, 2), System.Globalization.NumberStyles.HexNumber);
+                return new Color(new int[] { r, g, b });
+            } else if (token == "hsv") {
+                var hsv = new DoubleList(reader).Doubles;
+                if (hsv.Count != 3) {
+                    throw new Exception("Color has wrong number of components");
+                }
+                return new Color(new double[] { hsv[0], hsv[1], hsv[2] });
+            } else if (token == "hsv360") {
+                var hsv = new DoubleList(reader).Doubles;
+                if (hsv.Count != 3) {
+                    throw new Exception("Color has wrong number of components");
+                }
+                return new Color(new double[] { hsv[0] / 360, hsv[1] / 100, hsv[2] / 100 });
+            } else if (DoesRegexFullyMatch(new Regex(CommonRegexes.Catchall), token)) {
+                if (NamedColors.TryGetValue(token, out Color value)) {
+                    return value;
+                } else {
+                    throw new Exception(token + " was not a cached color");
+                }
+            } else {
+                var actualToken = token;
+                foreach(var ch in actualToken.ToCharArray().Reverse()) {
+                    reader.PushBack(ch);
+                }
+                var questionableList = new StringOfItem(reader).String;
+                if (questionableList.IndexOf('.') != -1) {
+                    // This is a double list.
+                    var doubleStreamReader = new BufferedReader(questionableList);
+                    var hsv = new DoubleList(doubleStreamReader).Doubles;
+                    if (hsv.Count != 3) {
+                        throw new Exception("Color has wrong number of components");
+                    }
+                    return new Color(new int[] { (int)Math.Round(hsv[0] * 255), (int)Math.Round(hsv[1] * 255), (int)Math.Round(hsv[2] * 255) });
+                } else {
+                    // integer list
+                    var integerStreamReader = new BufferedReader(questionableList);
+                    var rgb = new IntList(integerStreamReader).Ints;
+                    if (rgb.Count != 3) {
+                        throw new Exception("Color has wrong number of components");
+                    }
+                    return new Color(rgb.ToArray());
+                }
+            }
+        }
+
+        public Color GetColor(string colorName) {
+            if (NamedColors.TryGetValue(colorName, out Color value)) {
+                return value;
+            } else {
+                throw new Exception(colorName + " was not a cached color");
```

Comment:
Perhaps ArgumentException, but DataException seems a little better

Reply:
DataException sounds good, but it's purpose on Microsoft Docs doesn't fit here. I'll go with ArgumentException.